addons/io_hubs_addon/components/definitions/loop_animation.py

```diff
@@ -1,9 +1,160 @@
 import atexit
 import bpy
-from bpy.props import StringProperty, CollectionProperty, IntProperty, BoolProperty
+from bpy.app.handlers import persistent
+from bpy.props import StringProperty, CollectionProperty, IntProperty, BoolProperty, EnumProperty
 from bpy.types import PropertyGroup, Menu, Operator
 from ..hubs_component import HubsComponent
 from ..types import Category, PanelType, NodeType
+from ..utils import redraw_component_ui
+
+nla_track_name_msgbus_owner = None
+action_name_msgbus_owner = None
+
+class TrackPropertyType(PropertyGroup):
+    name: StringProperty(
+        name="Display Name",
+        description="Display Name",
+    )
+    track_name: StringProperty(
+        name="Track Name",
+        description="Track Name",
+    )
+    action_name: StringProperty( # Will only contain data if the track name is generic
+        name="Action Name",
+        description="Action Name",
+    )
+    track_type: EnumProperty(
+        name="Track Type",
+        description="Track Type",
+        items=[
+            ("object", "Object", "Object"),
+            ("shape_key", "Shape Key", "Shape Key")
+            ],
+        default="object"
+    )
+
+bpy.utils.register_class(TrackPropertyType)
+
+@atexit.register
+def unregister():
+    bpy.utils.unregister_class(TrackPropertyType)
+
+def register_msgbus():
+    global nla_track_name_msgbus_owner
+    global action_name_msgbus_owner
+
+    if nla_track_name_msgbus_owner:
+        return
+
+    nla_track_name_msgbus_owner = object()
+    bpy.msgbus.subscribe_rna(
+        key=(bpy.types.NlaTrack, "name"),
+        owner=nla_track_name_msgbus_owner,
+        args=(bpy.context,),
+        notify=redraw_component_ui,
+    )
+
+    if action_name_msgbus_owner:
+        return
+
+    action_name_msgbus_owner = object()
+    bpy.msgbus.subscribe_rna(
+        key=(bpy.types.Action, "name"),
+        owner=action_name_msgbus_owner,
+        args=(bpy.context,),
+        notify=redraw_component_ui,
+    )
+
+def unregister_msgbus():
+    global nla_track_name_msgbus_owner
+    global action_name_msgbus_owner
+
+    bpy.msgbus.clear_by_owner(nla_track_name_msgbus_owner)
+    bpy.msgbus.clear_by_owner(action_name_msgbus_owner)
+
+    nla_track_name_msgbus_owner = None
+    action_name_msgbus_owner = None
+
+@persistent
+def load_post(dummy):
+    unregister_msgbus()
+    register_msgbus()
+
+def is_default_name(track_name):
+    return bool(track_name.startswith("NlaTrack") or track_name.startswith("[Action Stash]"))
```

Comment:
I'm glad that we have tests as this is a good candidate for future breakage.

Reply:
Yes, definitely.